socorro/external/boto/connection_context.py

```diff
@@ -0,0 +1,245 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import boto
+import boto.s3.connection
+import boto.exception
+
+import json
+import os
+import socket
+import datetime
+import contextlib
+
+from configman import Namespace, RequiredConfig, class_converter
+from configman.converters import class_converter, py_obj_to_str
+from socorro.lib.util import DotDict
+from socorro.lib.converters import change_default
+
+class BotoNotFound(Exception):
+    pass
+
+
+#==============================================================================
+class ConnectionContext(RequiredConfig):
+
+    required_config = Namespace()
+    required_config.add_option(
+        'host',
+        doc="The hostname (leave empty for AWS)",
+        default="",
+        reference_value_from='resource.boto',
+    )
+    required_config.add_option(
+        'port',
+        doc="The network port (leave at 0 for AWS)",
+        default=0,
+        reference_value_from='resource.boto',
+    )
+    required_config.add_option(
+        'access_key',
+        doc="access key",
+        default="",
```

Comment:
we've not treated this as secret in the past.  To do so now, would require a change to configuration across the system where anything touches S3.